server/data/data-cassandra/src/test/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableTest.java

```diff
@@ -61,11 +79,60 @@ public static void tearDownClass() {
 
     @Override
     protected AbstractRecipientRewriteTable getRecipientRewriteTable() throws Exception {
-        CassandraRecipientRewriteTable rrt = new CassandraRecipientRewriteTable(
+        return buildRecipientRewriteTable(
             new CassandraRecipientRewriteTableDAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION),
-            new CassandraMappingsSourcesDAO(cassandra.getConf()));
+            new CassandraMappingsSourcesDAO(cassandra.getConf()),
+            new CassandraSchemaVersionDAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION));
+    }
+
+    private AbstractRecipientRewriteTable buildRecipientRewriteTable(CassandraRecipientRewriteTableDAO cassandraRecipientRewriteTableDAO,
+                                                                     CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO,
+                                                                     CassandraSchemaVersionDAO cassandraSchemaVersionDAO) throws Exception {
+        CassandraRecipientRewriteTable rrt = new CassandraRecipientRewriteTable(
+            cassandraRecipientRewriteTableDAO,
+            cassandraMappingsSourcesDAO,
+            cassandraSchemaVersionDAO);
         rrt.configure(new DefaultConfigurationBuilder());
         return rrt;
     }
 
+    @Test
+    public void listSourcesShouldUseDefaultMethodWhenMappingsSourcesIsUnsupported() throws Exception {
```

Comment:
We don't want specific tests IMO.

Maybe we do want to run `all tests` for version 6 and version 7